src/main/scala/scala/compat/java8/collectionImpl/Stepper.scala

```diff
@@ -0,0 +1,545 @@
+  * println(s.hasStep)                      //  Prints `false`
+  * }}}
+  */
+trait Stepper[@specialized(Double, Int, Long) A] extends StepperLike[A, Stepper[A]] {
+  /** Drains the contents of this stepper into an `Accumulator` or specialized variant thereof as appropriate.
+    * This is a terminal operation.
+    *
+    * Note: accumulation will occur sequentially.  To accumulate in parallel, use a `Stream` (i.e. `.parStream.accumulate`).
+    */
+  def accumulate[Acc <: AccumulatorLike[A, Acc]](implicit accer: scala.compat.java8.converterImpl.AccumulatesFromStepper[A, Acc]) = accer(this)  
+}
+
+/** An (optional) marker trait that indicates that a `Stepper` can call `substep` with
+  * at worst O(log N) time and space complexity, and that the division is likely to
+  * be reasonably even.
+  */
+trait EfficientSubstep {}
+
+/** Provides functionality for Stepper while keeping track of a more precise type of the collection.
+  */
+trait StepperLike[@specialized(Double, Int, Long) A, +CC] { self =>
+  /** Characteristics are bit flags that indicate runtime characteristics of this Stepper.
+    *
+    * - `Distinct` means that no duplicates exist
+    * - `Immutable` means that the underlying collection is guaranteed not to change during traversal
+    * - `NonNull` means that no nulls will be returned during traversal
+    * - `Sized` means that the collection knows its exact size
+    * - `SubSized` means that sub-Steppers created with `substep()` will also know their own size.  `SubSized` steppers must also be `Sized`.
+    *
+    * The Java flags `CONCURRENT` and `SORTED` are not supported; modification of a concurrency-aware underlying collection is not
+    * guaranteed to be any safer than modification of any generic mutable collection, and if the underlying collection is ordered by
+    * virtue of sorting, `Stepper` will not keep track of that fact.
+    */
+  def characteristics: Int
+
+  /** Returns the size of the collection, if known exactly, or `-1` if not. */
+  def knownSize: Long
+
+  /** `true` if there are more elements to step through, `false` if not. */
+  def hasStep: Boolean
+
+  /** The next element traversed by this Stepper.
+    * `nextStep()` throws an exception if no elements exist, so check `hasStep` immediately prior
+    * to calling.  Note that `tryStep` also consumes an element, so the result of `hasStep` will
+    * be invalid after `tryStep` is called.
+    */
+  def nextStep(): A
+
+  /** If another element exists, apply `f` to it and return `true`; otherwise, return `false`. */
+  def tryStep(f: A => Unit): Boolean
+
+  /** Attempt to split this `Stepper` in half, with the new (returned) copy taking the first half
+    * of the collection, and this one advancing to cover the second half.  If subdivision is not
+    * possible or not advisable, `substep()` will return `null`.
+    */
+  def substep(): CC
+
+  /** Returns the precise underlying type of this `Stepper`. */
+  def typedPrecisely: CC
```

Comment:
I don't remember, I'm afraid.  It was a bit tricky to get the manual specialization working and integrated properly with the automatic specialization, and it's possible that this was an artifact of my explorations there that ended up being necessary at some point (I am sure it was) but isn't any more.  Or maybe it was for Java interop?  I guess I could experiment with dropping it.